internal/pkg/node/constructors.go

```diff
@@ -47,39 +45,43 @@ func init() {
 	if DefaultConfig == "" {
 		DefaultConfig = path.Join(conf.Paths.Datadir, "warewulf/defaults.conf")
 	}
-	cachedDB.current = false
-	cachedDB.persist = true
 }
 
 /*
 Creates a new nodeDb object from the actual configuration
 */
 func New() (NodeYaml, error) {
-	if cachedDB.current {
-		wwlog.Debug("Returning cached object")
-		return cachedDB, nil
-	}
-	var ret NodeYaml
-
 	wwlog.Verbose("Opening node configuration file: %s", ConfigFile)
 	data, err := os.ReadFile(ConfigFile)
 	if err != nil {
-		return ret, err
+		return NodeYaml{}, err
 	}
+	return Parse(data)
+}
+
 
```

Comment:
Please add a comment what exactly Parse is doing and what kind of errors can occure